### PR TITLE
Milestone 4 - Category filter

### PR DESCRIPTION
**Pull request by Mupa1**
## Milestone 4 - Category filter

In this milestone, I added logic for filtering books by category. I:-
- Added CHANGE_FILTER action in actions/index.js
- Created filter reducer
- Added CategoryFilter component and rendered it in the BooksList component 
- Implemented filter logic in BooksList component

Kindly review my project.

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -7,3 +7,8 @@ export const REMOVE_BOOK = book => ({
   type: 'REMOVE_BOOK',
   payload: book.id,
 });
+
+export const CHANGE_FILTER = category => ({
```
Comment (MuriloRoque) [inline src/actions/index.js]: - [ ] Don't use UPPERCASE on your functions, as it's not a good practice, please switch to camelCase. Do the same for the other functions in this file.

File: src/containers/BooksList.js [hunk 2]
```diff
@@ -3,41 +3,67 @@ import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
 import Book from '../components/Book';
-import { REMOVE_BOOK } from '../actions/index';
+import { REMOVE_BOOK, CHANGE_FILTER } from '../actions/index';
```
Comment (MuriloRoque) [inline src/containers/BooksList.js]: - [ ] Don't use UPPERCASE on your functions, as it's not a good practice, please switch to camelCase.

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -3,41 +3,67 @@ import { connect } from 'react-redux';
...
+import CategoryFilter from '../components/CategoryFilter';
 
-const BooksList = ({ books, REMOVE_BOOK }) => {
+const BooksList = ({
+  books, REMOVE_BOOK, filter, CHANGE_FILTER,
```
Comment (MuriloRoque) [inline src/containers/BooksList.js]: - [ ] Don't use UPPERCASE on your functions, as it's not a good practice, please switch to camelCase.

File: src/containers/BooksList.js [hunk 4]
```diff
@@ -3,41 +3,67 @@ import { connect } from 'react-redux';
...
+}) => {
   const handleRemoveBook = book => {
     REMOVE_BOOK(book);
```
Comment (MuriloRoque) [inline src/containers/BooksList.js]: - [ ] Don't use UPPERCASE on your functions, as it's not a good practice, please switch to camelCase.

File: src/containers/BooksList.js [hunk 5]
```diff
@@ -3,41 +3,67 @@ import { connect } from 'react-redux';
...
   };
 
+  const handleFilterChange = e => {
+    CHANGE_FILTER(e.target.value);
```
Comment (MuriloRoque) [inline src/containers/BooksList.js]: - [ ] Don't use UPPERCASE on your functions, as it's not a good practice, please switch to camelCase.

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
## Status: Changes requested :red_circle: 

### Hello @Mupa1 :wave: 

### This milestone is done, everything is working as expected, congratulations! :clap: 

### I am only suggesting one small change, because of good practices, and I get it that you used UPPERCASE because of the action names, but remember that functions in JavaScript are written in camelCase :smile: Please check my comments in your code for the places you need to change.

Kindly make the requested changes and submit for another review, please :pray: 

I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Comment by Mupa1** [comment]
Thanks, @MuriloRoque for your feedback 😃 . I have made the requested changes 👍 .

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻

**Pull request merged**